### PR TITLE
feat: delete document api

### PR DESCRIPTION
**Pull request by grace0925**
Closes: #9 
Signed-off-by: Grace Liu <grace.liu@securekey.com>

### CODE REVIEW

File: pkg/restapi/operation/operations.go [hunk 1]
```diff
@@ -336,6 +339,34 @@ func (c *Operation) updateDocumentHandler(rw http.ResponseWriter, req *http.Requ
...
+//
+// Responses:
+//		default: genericError
+// 			200: emptyRes
+//			400: emptyRes
+// 			404: emptyRes
+func (c *Operation) deleteDocumentHandler(rw http.ResponseWriter, req *http.Request) {
+	vaultID, success := unescapePathVar(vaultIDPathVariable, mux.Vars(req), rw)
+	if !success {
+		return
```
Comment (Baha-sk) [inline pkg/restapi/operation/operations.go:354]: add log statement here to help troubleshoot this case

File: pkg/restapi/operation/operations.go [hunk 2]
```diff
@@ -336,6 +339,34 @@ func (c *Operation) updateDocumentHandler(rw http.ResponseWriter, req *http.Requ
...
+	}
+
+	docID, success := unescapePathVar(docIDPathVariable, mux.Vars(req), rw)
+	if !success {
+		return
```
Comment (Baha-sk) [inline pkg/restapi/operation/operations.go:359]: same here

**Review by fqutishat — CHANGES_REQUESTED** [review]
can we hold this change until we finish adding zcap to EDV server

File: pkg/restapi/operation/operations.go (hunk 1 above)
Comment (llorllale) [reply pkg/restapi/operation/operations.go:354]: To me this seems like it should return an error.

**Comment by grace0925** [comment]
> can we hold this change until we finish adding zcap to EDV server

yes

File: pkg/restapi/operation/operations.go (hunk 1 above)
Comment (grace0925) [reply pkg/restapi/operation/operations.go:354]: `unescapePathVar` was where the response was written. I have changed that to log the error and return BadRequest:

https://github.com/trustbloc/edv/pull/137/files#diff-5113c7f3feb6ad875b4dad132a94e674a7dbe720f0fa70d2b8ecee5c630191c3R24

**Review by fqutishat — APPROVED** [review]

File: pkg/restapi/operation/respond.go [hunk 3]
```diff
@@ -275,3 +275,18 @@ func writeUpdateDocumentFailure(rw http.ResponseWriter, errUpdateDoc error, docI
 		logger.Errorf(messages.UpdateDocumentFailure+messages.FailWriteResponse, docID, vaultID, errUpdateDoc, errWrite)
 	}
 }
+
+func writeDeleteDocumentFailure(rw http.ResponseWriter, errDeleteDoc error, docID, vaultID string) {
+	logger.Infof(messages.DeleteDocumentFailure, docID, vaultID, errDeleteDoc)
+
+	if errDeleteDoc == messages.ErrDocumentNotFound || errDeleteDoc == messages.ErrVaultNotFound {
```
Comment (sandrask) [inline pkg/restapi/operation/respond.go:282]: Is VaultNotFound considered 'bad request' or 'not found'?

**Comment by DRK3** [comment]
@fqutishat Will this change collide with your ZCAP stuff?

File: pkg/client/client_test.go [hunk 4]
```diff
@@ -482,6 +482,60 @@ func TestClient_UpdateDocument_ServerUnreachable(t *testing.T) {
...
+	validConfig := getTestValidDataVaultConfiguration()
+	vaultLocationURL, err := client.CreateDataVault(&validConfig)
+	require.NoError(t, err)
+
+	vaultID := getVaultIDFromURL(vaultLocationURL)
+
+	_, err = client.CreateDocument(vaultID, getTestValidEncryptedDocument(testJWE))
+	require.NoError(t, err)
+
+	err = client.DeleteDocument(vaultID, testDocumentID)
```
Comment (sandrask) [inline pkg/client/client_test.go:503]: Can you add a call to resolve deleted document here? I saw that you have this test at store level (great!) but it is also nice to have it at client level.

**Review by sandrask — APPROVED** [review]

File: pkg/restapi/operation/respond.go (hunk 3 above)
Comment (grace0925) [reply pkg/restapi/operation/respond.go:282]: I considered it as 'not found' since we do it this way in the readDocumentHandler

Comment (sandrask) [reply pkg/restapi/operation/respond.go:282]: sure, np

**Pull request merged**